src/Storages/System/StorageSystemZooKeeper.cpp

```diff
@@ -487,12 +558,24 @@ void ReadFromSystemZooKeeper::applyFilters(ActionDAGNodes added_filter_nodes)
 {
     SourceStepWithFilter::applyFilters(added_filter_nodes);
 
+    name = extractName(added_filter_nodes.nodes, context);
     paths = extractPath(added_filter_nodes.nodes, context, context->getSettingsRef()[Setting::allow_unrestricted_reads_from_keeper]);
 }
 
 
 Chunk SystemZooKeeperSource::generate()
 {
+    if (name.empty())
+    {
+        chassert(0); // In fact, it must always have a default value.
+        if (!started)
+            throw Exception(ErrorCodes::BAD_ARGUMENTS,
+                        "SELECT from system.zookeeper table must contain condition like zookeeperName = 'name'.");
```

Comment:
Consider using `LOGICAL_ERROR` error code: it works as `assert` in debug and sanitizer builds, and as a normal exception in release builds

Reply:
Also, we should not make this condition mandatory: if there's no condition on zookeeperName - it should use the default one (I understand that this code should be unreachable anyway, but the exception message is a bit confusing)

Reply:
Yes, that sounds reasonable. I'll leave the assert for dev and duplicate a default value for release.